2-extra/2-water-bottle.js

```diff
@@ -19,18 +19,55 @@ let bottle = {
   volume: 0,
   fillUp: function () {
     // calling this function should completely fill your bottle (volume = 100);
+      if (this.volume = 100){
+        return true
+      } else if(bottle.isEmpty){
+        return true
+      } else if(bottle.pour){
+        this.volume = 100
+      }
   },
   pour: function () {
     // calling this function should increase your bottle volume by 10 units;
+    if (bottle.isEmpty()){
+      this.volume = this.volume + 10
+    } 
   },
   drink: function () {
     // calling this function should decrease your bottle volume by 10 units;
+    if (bottle.isFull()){
+        this.volume = this.volume - 10
+      } else if (!bottle.isFull() && !bottle.isEmpty()){
+        this.volume = this.volume -10
+      } else if(bottle.isEmpty()){
+        this.volume = 0
+      }
   },
+
   isFull: function () {
     // this function should return true if your bottle is full;
+    if (this.volume == 100){
+      return true
+    } else if (this.volume < 100){
+      return false
+    } else if(this.volume > 100){
+      return false
+    } else if (bottle.pour()){
+      this.volume = 100
+    }
+  
   },
   isEmpty: function () {
     // this function should return true if your bottle is empty;
+    if (this.volume == 0){
```

Comment:
Same logic as the comment for isFull() here. If is empty (volume === 0) then is true, any other case it will return false.